plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py

```diff
@@ -82,6 +95,76 @@ def decode(
         return pl.read_parquet(uri, use_pyarrow=True, storage_options=kwargs)
 
 
+############################
+# Polars LazyFrame classes #
+############################
+
+class PolarsLazyFrameToParquetEncodingHandler(StructuredDatasetEncoder):
+    def __init__(self):
+        super().__init__(pl.LazyFrame, None, PARQUET)
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        lf = typing.cast(pl.LazyFrame, structured_dataset.dataframe)
+
+        output_bytes = io.BytesIO()
+        # The pl.LazyFrame.sink_parquet method uses streaming mode, which is currently considered unstable. Until it is
+        # stable, we collect the dataframe and write it to a BytesIO buffer.
+        df = lf.collect()
+        # Polars 0.13.12 deprecated to_parquet in favor of write_parquet
+        if hasattr(df, "write_parquet"):
+            df.write_parquet(output_bytes)
+        else:
+            df.to_parquet(output_bytes)
+
+        if structured_dataset.uri is not None:
+            fs = ctx.file_access.get_filesystem_for_path(path=structured_dataset.uri)
+            with fs.open(structured_dataset.uri, "wb") as s:
+                s.write(output_bytes)
```

Comment:
Can we write directly into the cloud storage and not use `io.BytesIO()`?

```python
with fs.open(structured_dataset.uri, "wb") as s:
    df.write_parquet(s)
```


REF: https://docs.pola.rs/user-guide/io/cloud-storage/